### ISSUE TITLE
Incorrect string marshaling when "ID3D12Object::SetName" was called

### ISSUE THREAD

**Issue by Cutano** [dotnet/Silk.NET#630]
# Summary
`SetName("DepthStencil")`
But I got
`(0x000001CAA011C1C0:'Dphtni)`

# Steps to reproduce
- Platform:Desktop
- Framework Version: .NET 6
- API: DirectX
- API Version: DirectX 12

```C#
ID3D12Resource* depthStencil;
.......... Some Code Here ............
depthStencil->SetName($"DepthStencil");
```
Then I got this error:
```
Execution (From D3D12): ID3D12CommandQueue::ExecuteCommandLists: An ID3D12Resource object (0x000001CAA011C1C0:'**Dphtni**) referenced in a command list ............
```

# Comments
It looks like Silk is using `Marshal.StringToHGlobalAnsi()` rather than `StringToBSTR()`, which drops half of the letters due to the encoding.

![image](https://user-images.githubusercontent.com/21150043/136657083-4abc8c1a-0563-48c0-a37d-719341a9c9d5.png)

![image](https://user-images.githubusercontent.com/21150043/136656836-7732c8fb-2cce-4df7-9f64-b198a96dac58.png)

**Comment by Cutano** [comment]
> So basically every second character makes it? I guess it's a problem with UTF16 vs UTF8, which indicates an annotation mistake. @Perksey assigning you for now, assuming this is BuildTools work?

Yes, the native API uses LPCWSTR to receive the name, which points to UTF-16 wchar.

```cpp
HRESULT SetName(
  LPCWSTR Name
);
```

**Comment by Perksey** [comment]
Will fix this for 2.10. Assuming that BuildTools already collects the `LPCWSTR` info, shouldn't be too much work...
